src/cascadia/TerminalCore/terminalrenderdata.cpp

```diff
@@ -27,23 +27,15 @@ const TextBuffer& Terminal::GetTextBuffer() noexcept
     return *_buffer;
 }
 
-// Creating a FontInfo can technically throw (on string allocation) and this is noexcept.
-// That means this will std::terminate. We could come back and make there be a default constructor
-// backup to FontInfo that throws no exceptions and allocates a default FontInfo structure.
-#pragma warning(push)
-#pragma warning(disable : 26447)
 const FontInfo& Terminal::GetFontInfo() noexcept
 {
-    // TODO: This font value is only used to check if the font is a raster font.
-    // Otherwise, the font is changed with the renderer via TriggerFontChange.
-    // The renderer never uses any of the other members from the value returned
-    //      by this method.
-    // We could very likely replace this with just an IsRasterFont method
-    //      (which would return false)
-    static const FontInfo _fakeFontInfo(DEFAULT_FONT_FACE, TMPF_TRUETYPE, 10, { 0, DEFAULT_FONT_SIZE }, CP_UTF8, false);
-    return _fakeFontInfo;
+    return _fontInfo;
+}
+
+void Terminal::SetFontInfo(const FontInfo& fontInfo)
+{
+    _fontInfo = fontInfo;
 }

```

Comment:
We specifically need this to get the font name (already in this PR) and the font size (excluded from this PR).

What's annoying is that this is a Terminal-specific issue too. ConHost returns an actual thing in `GetFontInfo`. Terminal, for whatever reason (we probably just didn't need it at the time tbh), lies.

> We'd need some sort of dummy implementation layer that mostly just calls into Terminal for the majority of the implementation, and ControlCore just for that one method... that seems dumb.

What if we double down on this info being stored in TerminalCore? And anytime ControlCore needs something from it, it has to call `_terminal.GetFontInfo()`? That way this is all in one place. (this would have to be a follow-up)

Reply:
> we probably just didn't need it at the time tbh

The need for it was solely to get the renderer to "do the right thing". The terminal actually didn't need to know anything about its font at any point other than _rendering_.

All it did was jog the renderer into knowing that it was, in fact, not rendering a raster font.

Why we cared about *that*, I will never know.